src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (!Truck.class.equals(type)) {
+            if (!Excavator.class.equals(type)) {
+                if (!Bulldozer.class.equals(type)) {
+                    return Collections.emptyList();
+                }
+                BulldozerProducer bulldozerProducer = new BulldozerProducer();
+                return new ArrayList<>(bulldozerProducer.get());
+            }
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            return new ArrayList<>(excavatorProducer.get());
+        }
+        TruckProducer truckProducer = new TruckProducer();
+        return new ArrayList<>(truckProducer.get());
+    }
```

Comment:
`BulldozerProducer, TruckProducer, ExcavatorProducer` must be class fields
```suggestion
    private final BulldozerProducer bulldozerProducer = new BulldozerProducer();
    private final TruckProducer truckProducer = new TruckProducer();
    private final ExcavatorProducer excavatorProducer = new ExcavatorProducer();
    
    @Override
    public List<Machine> getAll(Class<? extends Machine> type) {
        if (type == Bulldozer.class) {
            return new ArrayList<>(bulldozerProducer.get());
        } else if (type == Truck.class) {
            return new ArrayList<>(truckProducer.get());
        } else if (type == Excavator.class) {
            return new ArrayList<>(excavatorProducer.get());
        }
        return Collections.emptyList();
    }
```